### ISSUE TITLE
No audio when playing some mkv container format files

### ISSUE THREAD

**Issue by Goblinlordx** [jellyfin/jellyfin-androidtv#1393]
### Describe the bug

1. Go to 'Movies"
2. Attempt to play certain format movies
3. No audio is heard

Seems to occur primarily on mkv file formats. Not sure about other formats. These are always "direct plays" and they work on the most recent `jellyfin` client on an Android phone. These videos also seem to work via web although not by "direct play" but with transcoding.

### Media info of the file

```shell
Stream #0:0(eng): Video: h264 (High), yuv420p(tv, bt709, progressive), 1920x1080 [SAR 1:1 DAR 16:9], 24 fps, 24 tbr, 1k tbn, 48 tbc (default)
    Metadata:
      BPS             : 6205044
      DURATION        : 00:37:40.209000000
      NUMBER_OF_FRAMES: 54245
      NUMBER_OF_BYTES : 1753087130
      _STATISTICS_WRITING_APP: mkvmerge v64.0.0 ('Willows') 64-bit
      _STATISTICS_TAGS: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES
    Stream #0:1(eng): Audio: eac3, 48000 Hz, 6 channels, fltp (default)
    Metadata:
      BPS             : 768000
      DURATION        : 00:37:40.224000000
      NUMBER_OF_FRAMES: 70632
      NUMBER_OF_BYTES : 216981504
      _STATISTICS_WRITING_APP: mkvmerge v64.0.0 ('Willows') 64-bit
      _STATISTICS_TAGS: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES
    Stream #0:2(eng): Subtitle: subrip
    Metadata:
      title           : English (SDH)
      BPS             : 29
      DURATION        : 00:33:30.709000000
      NUMBER_OF_FRAMES: 261
      NUMBER_OF_BYTES : 7375
      _STATISTICS_WRITING_APP: mkvmerge v64.0.0 ('Willows') 64-bit
      _STATISTICS_TAGS: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES
    Stream #0:3(dan): Subtitle: subrip
    Metadata:
      title           : Dansk
      BPS             : 20
      DURATION        : 00:30:45.000000000
      NUMBER_OF_FRAMES: 105
      NUMBER_OF_BYTES : 4735
      _STATISTICS_WRITING_APP: mkvmerge v64.0.0 ('Willows') 64-bit
      _STATISTICS_TAGS: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES
    Stream #0:4(ger): Subtitle: subrip
    Metadata:
      title           : Deutsch
      BPS             : 19
      DURATION        : 00:34:43.709000000
      NUMBER_OF_FRAMES: 144
      NUMBER_OF_BYTES : 5059
      _STATISTICS_WRITING_APP: mkvmerge v64.0.0 ('Willows') 64-bit
      _STATISTICS_TAGS: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES
    Stream #0:5(spa): Subtitle: subrip
    Metadata:
      title           : Español (Latinoamericano)
      BPS             : 17
      DURATION        : 00:34:43.709000000
      NUMBER_OF_FRAMES: 146
      NUMBER_OF_BYTES : 4633
      _STATISTICS_WRITING_APP: mkvmerge v64.0.0 ('Willows') 64-bit
      _STATISTICS_TAGS: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES
    Stream #0:6(spa): Subtitle: subrip
    Metadata:
      title           : Español
      BPS             : 17
      DURATION        : 00:34:43.709000000
      NUMBER_OF_FRAMES: 146
      NUMBER_OF_BYTES : 4636
      _STATISTICS_WRITING_APP: mkvmerge v64.0.0 ('Willows') 64-bit
      _STATISTICS_TAGS: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES
    Stream #0:7(fin): Subtitle: subrip
    Metadata:
      title           : Suomi
      BPS             : 18
      DURATION        : 00:34:43.709000000
      NUMBER_OF_FRAMES: 145
      NUMBER_OF_BYTES : 4874
      _STATISTICS_WRITING_APP: mkvmerge v64.0.0 ('Willows') 64-bit
      _STATISTICS_TAGS: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES
    Stream #0:8(fre): Subtitle: subrip
    Metadata:
      title           : Français (Canadien)
      BPS             : 21
      DURATION        : 00:27:57.333000000
      NUMBER_OF_FRAMES: 151
      NUMBER_OF_BYTES : 4466
      _STATISTICS_WRITING_APP: mkvmerge v64.0.0 ('Willows') 64-bit
      _STATISTICS_TAGS: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES
    Stream #0:9(fre): Subtitle: subrip
    Metadata:
      title           : Français
      BPS             : 17
      DURATION        : 00:34:43.792000000
      NUMBER_OF_FRAMES: 152
      NUMBER_OF_BYTES : 4522
      _STATISTICS_WRITING_APP: mkvmerge v64.0.0 ('Willows') 64-bit
      _STATISTICS_TAGS: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES
    Stream #0:10(ita): Subtitle: subrip
    Metadata:
      title           : Italiano
      BPS             : 18
      DURATION        : 00:34:43.709000000
      NUMBER_OF_FRAMES: 145
      NUMBER_OF_BYTES : 4859
      _STATISTICS_WRITING_APP: mkvmerge v64.0.0 ('Willows') 64-bit
      _STATISTICS_TAGS: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES
    Stream #0:11(jpn): Subtitle: subrip
    Metadata:
      title           : Japanese
      BPS             : 17
      DURATION        : 00:37:13.125000000
      NUMBER_OF_FRAMES: 139
      NUMBER_OF_BYTES : 4880
      _STATISTICS_WRITING_APP: mkvmerge v64.0.0 ('Willows') 64-bit
      _STATISTICS_TAGS: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES
    Stream #0:12(kor): Subtitle: subrip
    Metadata:
      title           : Korean
      BPS             : 17
      DURATION        : 00:37:13.125000000
      NUMBER_OF_FRAMES: 147
      NUMBER_OF_BYTES : 4906
      _STATISTICS_WRITING_APP: mkvmerge v64.0.0 ('Willows') 64-bit
      _STATISTICS_TAGS: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES
    Stream #0:13(dut): Subtitle: subrip
    Metadata:
      title           : Nederlands
      BPS             : 18
      DURATION        : 00:34:43.709000000
      NUMBER_OF_FRAMES: 137
      NUMBER_OF_BYTES : 4900
      _STATISTICS_WRITING_APP: mkvmerge v64.0.0 ('Willows') 64-bit
      _STATISTICS_TAGS: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES
    Stream #0:14(nor): Subtitle: subrip
    Metadata:
      title           : Norsk
      BPS             : 16
      DURATION        : 00:34:43.709000000
      NUMBER_OF_FRAMES: 145
      NUMBER_OF_BYTES : 4423
      _STATISTICS_WRITING_APP: mkvmerge v64.0.0 ('Willows') 64-bit
      _STATISTICS_TAGS: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES
    Stream #0:15(por): Subtitle: subrip
    Metadata:
      title           : Português
      BPS             : 17
      DURATION        : 00:34:43.709000000
      NUMBER_OF_FRAMES: 146
      NUMBER_OF_BYTES : 4565
      _STATISTICS_WRITING_APP: mkvmerge v64.0.0 ('Willows') 64-bit
      _STATISTICS_TAGS: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES
    Stream #0:16(por): Subtitle: subrip
    Metadata:
      title           : Português (Brasil)
      BPS             : 18
      DURATION        : 00:34:43.709000000
      NUMBER_OF_FRAMES: 146
      NUMBER_OF_BYTES : 4916
      _STATISTICS_WRITING_APP: mkvmerge v64.0.0 ('Willows') 64-bit
      _STATISTICS_TAGS: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES
    Stream #0:17(swe): Subtitle: subrip
    Metadata:
      title           : Svenska
      BPS             : 20
      DURATION        : 00:30:45.000000000
      NUMBER_OF_FRAMES: 120
      NUMBER_OF_BYTES : 4685
      _STATISTICS_WRITING_APP: mkvmerge v64.0.0 ('Willows') 64-bit
      _STATISTICS_TAGS: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES
    Stream #0:18(chi): Subtitle: subrip
    Metadata:
      title           : Chinese (Traditional)
      BPS             : 12
      DURATION        : 00:37:13.125000000
      NUMBER_OF_FRAMES: 147
      NUMBER_OF_BYTES : 3607
      _STATISTICS_WRITING_APP: mkvmerge v64.0.0 ('Willows') 64-bit
      _STATISTICS_TAGS: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES
    Stream #0:19(chi): Subtitle: subrip
    Metadata:
      title           : Chinese (Simplified)
      BPS             : 13
      DURATION        : 00:37:13.125000000
      NUMBER_OF_FRAMES: 147
      NUMBER_OF_BYTES : 3847
      _STATISTICS_WRITING_APP: mkvmerge v64.0.0 ('Willows') 64-bit
      _STATISTICS_TAGS: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES
    Stream #0:20(chi): Subtitle: subrip
    Metadata:
      title           : Chinese (Hong Kong)
      BPS             : 13
      DURATION        : 00:37:13.125000000
      NUMBER_OF_FRAMES: 147
      NUMBER_OF_BYTES : 3904
      _STATISTICS_WRITING_APP: mkvmerge v64.0.0 ('Willows') 64-bit
      _STATISTICS_TAGS: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES
```
```


### Application version

0.12.3

### Device information

Chromecast with Google TV

### Android version

Android 10

### Jellyfin server version

10.7.7

**Comment by hlbstd** [comment]
I reproduce the same behavior when playing files with eac3 audio in exoplayer. Enabling server side audio transcoding or using libvlc works around this issue but both methods have the undesired consequence of breaking dolby vision playback.

### Application version

0.12.3

### Device information

NVIDIA SHIELD Android TV

### Android version

Android 11

### Jellyfin server version

10.7.7

**Comment by DavidFair** [comment]
eac3 isn't natively supported by [Exoplayer](https://exoplayer.dev/supported-formats.html#ffmpeg-extension)
But this should be fixed by #1369 - can you give it a test once that PR gets merged (or even better before) and comment/close if this gets your audio working?

**Comment by Goblinlordx** [comment]
Not entirely sure about the support here. Either way, actually this seems to be working on `v0.13.0-beta.1`. I will close this as resolved regarding my issue.
